src/main/java/com/conveyal/analysis/controllers/OpportunityDatasetController.java

```diff
@@ -140,29 +141,31 @@ private OpportunityDatasetUploadStatus downloadLODES(Request req, Response res)
         final String regionId = req.params("regionId");
         final int zoom = parseZoom(req.queryParams("zoom"));
 
-        // default
-        final String accessGroup = req.attribute("accessGroup");
-        final String email = req.attribute("email");
-        final Region region = Persistence.regions.findByIdIfPermitted(regionId, accessGroup);
+        UserPermissions userPermissions = req.attribute(USER_PERMISSIONS_ATTRIBUTE);
+        final Region region = Persistence.regions.findByIdIfPermitted(regionId, userPermissions.accessGroup);
         // Common UUID for all LODES datasets created in this download (e.g. so they can be grouped together and
         // deleted as a batch using deleteSourceSet)
-        final String downloadBatchId = new ObjectId().toString();
         // The bucket name contains the specific lodes data set and year so works as an appropriate name
         final OpportunityDatasetUploadStatus status = new OpportunityDatasetUploadStatus(regionId, extractor.sourceName);
         addStatusAndRemoveOldStatuses(status);
 
-        taskScheduler.enqueueHeavyTask(() -> {
-            try {
-                status.message = "Extracting census data for region";
-                List<Grid> grids = extractor.censusDataForBounds(region.bounds, zoom);
-                createDatasetsFromPointSets(
-                        email, accessGroup, extractor.sourceName, downloadBatchId, regionId, status, grids
-                );
-            } catch (IOException e) {
-                status.completeWithError(e);
-                LOG.error("Exception processing LODES data: " + ExceptionUtils.stackTraceString(e));
-            }
-        });
+        SpatialResource source = SpatialResource.create(userPermissions, extractor.sourceName)
```

Comment:
This will create a new distinct SpatialResource entity each time someone imports from LODES. If I'm not mistaken we will end up with one entity per region, and multiple entities per region if for whatever reason the import operation is repeated (e.g. failure or deletion of resulting layers, then re-import). It doesn't seem right to have multiple entities all referencing the same single data source (the seamless census dataset for a particular year). Also, the name field in the new entity contains the human-readable name of the LODES data set, while on other entities of this type it seems like it might contain a filename.

Reply:
In today's call we decided to create one stand-in instance of each Conveyal-managed data source per region, and re-use that instance if multiple import operations occur in the same region. This will have the desired effect of grouping all LODES layers from the same year together, with sub-groupings for each import operation, using only general purpose code and avoiding some special case where a single application-wide entity (lacking region ID and group affiliation) represents the Conveyal-managed data source.